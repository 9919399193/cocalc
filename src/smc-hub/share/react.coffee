###
Load react support for rendering.
###

async = require('async')

# Code for rendering react components to html.
ReactDOMServer = require('react-dom/server')

require('./jsdom-support')

# Uncomment for cc-in-cc dev benchmarking purposes.  This variable is already set
# by the Docker container when running in kubernetes.
## process.NODE_ENV="production"

require('smc-webapp/r_misc').SHARE_SERVER = true

# Load katex jQuery plugin.
require('smc-webapp/jquery-plugins/katex')

exports.react = (res, component, extra, viewer) ->
    res.type('html')
    t0 = new Date()
    stream = ReactDOMServer.renderToStaticNodeStream(component)
    stream.pipe(res)
    stream.once 'end', ->
        console.log("react: time to render and stream out: #{new Date() - t0}ms", extra)
