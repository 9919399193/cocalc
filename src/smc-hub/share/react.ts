/*
Load react support for rendering.
*/

// Code for rendering react components to html.
import * as ReactDOMServer from "react-dom/server";

import "./jsdom-support";

// Uncomment for cc-in-cc dev benchmarking purposes.  This variable is already set
// by the Docker container when running in kubernetes.
//# process.NODE_ENV="production"

import { set_share_server } from "smc-webapp/r_misc/share-server";

set_share_server(true);

// Load katex jQuery plugin.
require("smc-webapp/jquery-plugins/katex");

export function react(res, component, extra: any = ""): void {
  res.type("html");
  const t0 = new Date().valueOf();
  const stream = ReactDOMServer.renderToStaticNodeStream(component);
  stream.pipe(res);
  stream.once("end", () =>
    console.log(
      `react: time to render and stream out: ${new Date().valueOf() - t0}ms`,
      extra
    )
  );
}
