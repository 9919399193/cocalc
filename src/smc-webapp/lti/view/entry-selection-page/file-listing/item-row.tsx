import styled from "styled-components";
import { default_colors } from "../../values";

interface Props {
  highlight: boolean;
}

export const ItemRow = styled.div<Props>`
  cursor: pointer
  color: ${props => (props.highlight ? "SeaGreen" : `${default_colors.color}`)};
  margin: 2px;
`;
