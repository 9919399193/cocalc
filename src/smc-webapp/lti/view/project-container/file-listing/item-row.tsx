import styled from "styled-components";

interface Props {
  highlight: boolean;
}

export const ItemRow = styled.div<Props>`
  cursor: pointer
  color: ${props => (props.highlight ? "ForestGreen" : "Black")};
  margin: 2px;
`;
