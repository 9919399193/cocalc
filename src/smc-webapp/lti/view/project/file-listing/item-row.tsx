import styled from "styled-components";

interface Props {
  highlight: boolean;
}

export const ItemRow = styled.div<Props>`
  cursor: pointer
  color: ${props => (props.highlight ? "ForestGreen" : "DarkSlateBlue")};
  margin: 2px;
`;
