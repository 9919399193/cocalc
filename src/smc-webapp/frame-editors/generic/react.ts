// THIS SHOULD BE MOVED OUT OF frame-editors/


export const {
  React,
  ReactDOM,
  rclass,
  rtypes,
  Fragment,
  redux,
  Redux
} = require("smc-webapp/smc-react");

import * as ReactOrig from "react";

export const Component = ReactOrig.Component;

export type Rendered = ReactOrig.ReactElement<any> | undefined;
