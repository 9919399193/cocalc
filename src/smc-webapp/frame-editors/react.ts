export const {
  React,
  ReactDOM,
  rclass,
  rtypes,
  Fragment,
  redux,
  Redux
} = require("smc-webapp/smc-react");

import * as ReactOrig from "react";

export const Component = ReactOrig.Component;

export type Rendered = ReactOrig.ReactElement<any> | undefined;
