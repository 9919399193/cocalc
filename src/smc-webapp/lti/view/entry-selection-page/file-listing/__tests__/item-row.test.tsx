import * as React from "react";
import { shallow } from "enzyme";
import { ItemRow } from "../item-row";

describe("Render regressions", () => {
  test("highlight == true", () => {
    const rendered = shallow(<ItemRow highlight={true} />);

    expect(rendered).toMatchSnapshot("highlight color is SeaGreen");
  });

  test("highlight == false", () => {
    const rendered = shallow(<ItemRow highlight={false} />);

    expect(rendered).toMatchSnapshot("highlight color is not SeaGreen");
  });
});
