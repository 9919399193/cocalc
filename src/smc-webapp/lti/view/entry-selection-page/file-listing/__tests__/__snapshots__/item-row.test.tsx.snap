// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render regressions highlight == false: highlight color is not SeaGreen 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  cursor: pointer
  color: ",
          [Function],
          ";
  margin: 2px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight={false}
/>
`;

exports[`Render regressions highlight == true: highlight color is SeaGreen 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  cursor: pointer
  color: ",
          [Function],
          ";
  margin: 2px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight={true}
/>
`;
